### ISSUE TITLE
FABL links to finding aids

### ISSUE THREAD

**Issue by marquis** [cul/fabl#2]
FABL records link back to full finding aid displays.
- Make these links work for CUL finding aids
- Link to sub-components of CUL finding aids

**Comment by marquis** [comment]
The link is built by application_helper.rb.  See:
link_to_findingaid()
get_url_for_findingaid_from_document()
url_for_findingaid()

I've begun work to change these over for Columbia.

What different levels of links do we need to generate?

top-level:   
http://findingaids.cul.columbia.edu/ead/nnc-rb/ldpd_4078917
container-list:  
http://findingaids.cul.columbia.edu/ead/nnc-rb/ldpd_4078917/dsc
series:
http://findingaids.cul.columbia.edu/ead/nnc-rb/ldpd_4078917/dsc/5
subseries:
http://findingaids.cul.columbia.edu/ead/nnc-rb/ldpd_4078917/dsc/5#subseries4

How do the fields in the Solr document help us build these paths?

format_sim:
682,491  Archival Object
9,915  Archival Series
1,232  Archival Collection

level_sim
675,588    file
5,282    item
5,062    series
4,853    subseries
1,505    otherlevel
59    subgrp

repository_sim
505,196    nnc-rb
138,598    nnc-ua
47,247    nnc-a
2,597   nnc-ut

**Comment by marquis** [comment]
I've reworked the code to get to the root of finding aids correctly, but not the component/subcomponent level.  Which fields do I look at to get this?  Here's a peek at the raw Solr data I have to work with.

http://localhost:8984/solr/fabl_dev/select?q=*&rows=10&fl=*&wt=json&indent=true&defType=edismax&qf=unittitle_teim&pf=unittitle_teim&stopwords=true&lowercaseOperators=true&qt=search&facet=off&spellcheck=off

Also, I need to get the "ldpd_999" component of the path.  Currently, I'm doing this:
   doc[:ead_ssi].sub('_ead.xml', '')
which I really don't like.  Any better ideas?

**Comment by tcatapano** [comment]
The collection id portion of the URL is 'ldpd_' concatenated with the clio num of the collection. The clio number is in the collection level unitid[@type = 'clio']. Im not sure how this is indexed into solr, however. Im not sure they distinguish types of unitids, and they certainly wouldnt know what the 'clio' type is anyway.  How can I take a look at the solr documents? I cant seem to access http://fabl-dev.cul.columbia.edu:8984/solr or http://128.59.241.63:8984/solr/

**Comment by marquis** [comment]
Whoops, that was my tunneling URL.   
For on-campus, this should work:

http://all-index-solr-test1.cul.columbia.edu:8983/solr/fabl_dev/select?q=&rows=10&fl=&wt=json&indent=true&defType=edismax&qf=unittitle_teim&pf=unittitle_teim&stopwords=true&lowercaseOperators=true&qt=search&facet=off&spellcheck=off
 
If it doesn't let me know.

**Comment by tcatapano** [comment]
the value is in collection_unitid_ssm, which in the results display is labelled "Call no" (we need to change that). In the case of many RBML and possibly other non-Avery finding aids, there will be two collection level unitids, so the values may be in a list

**Comment by marquis** [comment]
Hmmm, doesn't look like I can trust collection_unitid_ssm.  I'll just use the substring logic - it's reliable.

The UnitID parsing isn't so simple.  Archival Series records have a collection_unitid_ssm.  Archival Collection records have a unitid_ssm.  I can do the logic to differentiate these, but then I may have multiple Unit IDs, and don't know which from the list to take.

Here's a simple EAD:
    https://github.com/cul/FABL-Data/blob/master/ead/nnc-rb/ldpd_6915026_ead_fabl.xml

It's got two Unit Ids:  MS#1616 and 6915026.  Once these are translated into Solr fields, the 'type' is lost, so I'd have to do something like a regexp against /^\d+$/ to detect clio numbers.  That starts to feel as hacky as doing a substring on the original filename.

Here's the resulting Solr records for the Marie Read Smith EAD:

http://all-index-solr-test1.cul.columbia.edu:8983/solr/fabl_dev/select?q=Marie+Read+Smith+Papers&rows=10&fl=&wt=json&indent=true&defType=edismax&qf=unittitle_teim&pf=unittitle_teim&stopwords=true&lowercaseOperators=true&qt=search&facet=off&spellcheck=off

**Comment by marquis** [comment]
Ok, leaving alone bib-id extraction, how about sublevel determination?

Looking at this simple 6915026 example, I can query fabl-dev for "Marie Read Smith Papers" and see several records - one for the Archival Collection, one for the Archival Series, and three for Archival Objects

https://fabl-dev.cul.columbia.edu/catalog?client_id=&f%5Bcollection_sim%5D%5B%5D=Marie+Read+Smith%0A++++++++++++Papers%2C&login_path=%2Flogin&return_uri=https%3A%2F%2Ffabl-dev.cul.columbia.edu%2F

Or, direct Solr link:

http://all-index-solr-test1.cul.columbia.edu:8983/solr/fabl_dev/select?q="Mary+Read+Smith+Papers"&qf=unittitle_teim+parent_unittitles_teim&facet=off&spellcheck=off&fl=*&defType=edismax&wt=json

Currently, all levels link to the finding aid landing page.

As a first step, I can link the sub-levels (Series and Object) to the "/dsc" link - that'll help.

I'm going to scour the raw Solr field returned to see if I can find reliable indicators to reach the correct series/subseries.

**Comment by marquis** [comment]
No, I don't see anything I can do in there, other than regexp against the Unit Title - and I really don't want to start doing that.

I'll link to the /dsc for now.  Let me know if you have any ideas.

**Comment by tcatapano** [comment]
Yeah, this is why Im increasingly leaning towards putting explicit URIs in a unitid for each unit and sticking it into an anchor tag in the HTML rendering of the finding aid.

**Comment by tcatapano** [comment]
I think the thing to do is for now is just to link any object below series (subseries, file, item) to the parent series. We may be able to do the same with subseries, but since series arent as universal as series, it may be harder.

**Comment by marquis** [comment]
OK, everything but parent is going to /dsc - the container list.  This'll be deployed to the server environment shortly.  I'll let this ticket be for now.
